avm/res/machine-learning-services/workspace/tests/e2e/ai/main.test.bicep

```diff
@@ -0,0 +1,82 @@
+targetScope = 'subscription'
+
+metadata name = 'Creating Azure AI Studio resources'
+metadata description = 'This instance deploys an Azure AI hub workspace.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-machinelearningservices.workspaces-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'mlswai'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
```

Comment:
If purge protection is enabled, we need to ensure a unique name for the dependency is generated every time. This is usually done by adding a random suffix to it.

Please check on other kvnames created as dependencies to follow the same approach

Reply:
Thanks, added it to all key vaults in the module